src/os/errors.go

```diff
@@ -17,8 +18,46 @@ var (
 	ErrExist          = errors.New("file exists")
 )
 
+// IsExist returns a boolean indicating whether the error is known to report
+// that a file or directory already exists. It is satisfied by ErrExist as
+// well as some syscall errors.
+func IsExist(err error) bool {
+	return underlyingErrorIs(err, ErrExist)
+}
+
+// IsNotExist returns a boolean indicating whether the error is known to
+// report that a file or directory does not exist. It is satisfied by
+// ErrNotExist as well as some syscall errors.
+func IsNotExist(err error) bool {
+	return underlyingErrorIs(err, ErrNotExist)
+}
+
 func IsPermission(err error) bool {
-	return err == ErrPermission
+	return underlyingErrorIs(err, ErrPermission)
+}
+
+func underlyingErrorIs(err, target error) bool {
+	// Note that this function is not errors.Is:
+	// underlyingError only unwraps the specific error-wrapping types
+	// that it historically did, not all errors implementing Unwrap().
+	err = underlyingError(err)
+	if err == target {
+		return true
+	}
+	// To preserve prior behavior, only examine syscall errors.
+	e, ok := err.(syscall.Errno)
+	return ok && e.Is(target)
```

Comment:
Looks like `e.Is` causes build failure in go 1.12 and 1.11. If we want to resolve this, the implementation may be complicated. So I'm just curious how long TinyGo will support (at least on CI testing) 1.12 and 1.11? AFAIK, official Go's support on these versions are no longer available according to the official doc saying `Each major Go release is supported until there are two newer major releases.`  https://golang.org/doc/devel/release.html#policy .

I need maintainers input. Thanks in advance.

Reply:
That is a very valid point to consider. I just created https://github.com/tinygo-org/tinygo/issues/1672 to discuss more broadly.

Reply:
Thanks, I'm going to wait until the next release is published :)

Reply:
https://github.com/tinygo-org/tinygo/pull/1659/commits/e3bbae5fba2803027036e93ccfa6f91c014d38ae